### ISSUE TITLE
How to update to the latest firmware on a Raspberry Pi 1 B+ ?

### ISSUE THREAD

**Issue by nmonet** [maxnet/berryboot#196]
On the last version (20150808), you say that there is a new firmware version for the Raspberry Pi.
Is the firmware (not the kernel) only available for the Raspberry Pi 2 ?
If it is available for all Pi version, how to update it ? Do we need to recompile a version (for the Pi 1), or is it possible to take some files from the berryboot-20150808-pi2-only.zip ?

Thanks in advance.

**Comment by maxnet** [comment]
You can update the firmware by going to "add OS" and accepting the update.
However be aware that in the newer Raspberry Pi firmware CMA is broken. Requiring an extra reboot if you change to another OS which has a different memory split than the previous one.

**Comment by nmonet** [comment]
Thanks for the quick answer but I think I miss something.
I already have an running image (Raspbian version ?) using a ISCI LUN (Synology NAS). 
If I choose the "add OS " option, do I need to reinstall all software in the previous image, or the "add OS" update the current running image (without losing installed tools) ?
I understand for the CMA settings.

Thanks,

**Comment by maxnet** [comment]
If you press the "add OS" button it will trigger a download of the list of available operating systems.
The meta data of that list also mentions the latest Berryboot version number.
As soon as Berryboot figures out it's not running the latest version, it will give you popup asking if you want to update.

**Comment by nmonet** [comment]
I've tried to update the firmware of my Raspberry Pi 1, but unfortunately it doesn't works.
If I use of the "Add OS" option, it ask me to update the Raspberry Pi 2 (not a version 1 like I have).
If I click yes (even if it seems to 'detect' the wrong version of the Raspberry), it start updating ... but when I reboot, I doesn't work (the SD card files seems to have changed ?).

If you have other ideas of what's going wrong ?

**Comment by maxnet** [comment]
Define "doesn't work".
Hangs on blank screen? Rainbow screen? Error message?

**Comment by nmonet** [comment]
Then I reboot ( after the click to the close button of the update process), the boot occured, the berryboot menu appears, but an error message box appears ( unable to find the server, I think it is the iscsi server). 
Please note that I use the network adapter ( no wifi ).

**Comment by maxnet** [comment]
Ok, so it doesn't get along with the new firmware for reasons unkown.
You can get the old firmware back by extracting the old Berryboot .zip. Do not overwrite cmdline.txt

**Comment by nmonet** [comment]
Yes, I've revert back to the previous firmware ( I have made a copy of the sd before running the update ).
